### PR TITLE
feat: new onboarding event analytics

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- All analytics for when the user has selected some topics/tags are inside the hook it uses to update the local data (meaning, no additional tracking needed).
- Applied tracking of impression for the new onboarding version modal.
- Also applied tracking when the auth modal opens from the generic component named `SteppedModal`, which parameters will be coming from the props.
- Will update the taxonomy sheet once we verify the correct tracking events.

Tagging @idoshamun for review to ensure all requirements for events are met since it was mentioned in the spec to there be no additional ones.
[Reference](https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/198049793/Feed+filters#New-events):
image

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-670 #done

### CODE REVIEW

File: packages/shared/src/components/modals/SteppedModal.tsx [hunk 1]
```diff
@@ -117,6 +120,12 @@ function SteppedModal({
       return onFinish?.();
     }
 
+    trackEvent({
+      event_name: 'click',
+      target_type: trigger,
+      target_id: targetId,
+    });
```
Comment (rebelchris) [inline packages/shared/src/components/modals/SteppedModal.tsx:127]: Which one is this in the taxonomy?
I also don't see it in the changed/new flow

File: packages/shared/src/components/modals/OnboardingModal.tsx [hunk 2]
```diff
@@ -127,6 +138,15 @@ function OnboardingModal({
     }
   };
 
+  useEffect(() => {
+    trackEvent({
+      event_name: 'impression',
+      target_type: 'onboarding modal',
+      target_id: 'v2',
+      extra: JSON.stringify({ origin: mode }),
+    });
```
Comment (rebelchris) [inline packages/shared/src/components/modals/OnboardingModal.tsx:147]: This is not inline with the proposed taxonomy:

https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0

(My feed onboarding)

**Review by rebelchris — CHANGES_REQUESTED** [review]
I think you may have missed some properties.

I also see "my feed onboarding skip" in the taxonomy, could you ensure that's still there?

**Comment by idoshamun** [comment]
@sshanzel I'll wait with my review until we conclude the other PR

File: packages/shared/src/components/modals/OnboardingModal.tsx [hunk 3]
```diff
@@ -140,6 +160,7 @@ function OnboardingModal({
       invalidMessage={invalidMessage}
       isFirstStepIntroduction
       isLastStepLogin
+      targetId="onboarding-v2"
```
Comment (el) [inline packages/shared/src/components/modals/OnboardingModal.tsx:163]: Can we use an enum here?

File: packages/shared/src/components/modals/OnboardingModal.tsx [hunk 4]
```diff
@@ -127,6 +138,15 @@ function OnboardingModal({
     }
   };
 
+  useEffect(() => {
+    trackEvent({
+      event_name: 'impression',
```
Comment (el) [inline packages/shared/src/components/modals/OnboardingModal.tsx:143]: I think we should use enums for the values

**Review by el — APPROVED** [review]

File: packages/shared/src/components/modals/OnboardingModal.tsx (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/components/modals/OnboardingModal.tsx:143]: Totally agree with both of you.

File: packages/shared/src/components/modals/OnboardingModal.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/modals/OnboardingModal.tsx]: Thank you for pointing this out, I have now a better understanding of the taxonomy sheet.

**Comment by sshanzel** [comment]
I've accidentally merged the changes. Let me re-open a new PR. Keyboard issues 🤦

**Pull request merged**